File: e2e/elixir/test/map_test.exs
# E2e tests for category: map
defmodule E2e.MapTest do
  use ExUnit.Case, async: true

  describe "map_discover_urls" do
    test "Discovers all URLs on a site without fetching full content" do
      result = Kreuzcrawl.scrape!()
      assert length(result.urls) >= 3
    end
  end

  describe "map_exclude_patterns" do
    test "Excludes URLs matching patterns from URL map" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(length(result.urls)) == 1
    end
  end

  describe "map_include_subdomains" do
    test "Includes subdomain URLs in URL map discovery" do
      result = Kreuzcrawl.scrape!()
      assert length(result.urls) >= 2
      assert String.contains?(result.urls, "blog.example.com")
    end
  end

  describe "map_large_sitemap" do
    test "Handles large sitemap with 100+ URLs" do
      result = Kreuzcrawl.scrape!()
      assert length(result.urls) >= 100
    end
  end

  describe "map_limit_pagination" do
    test "Limits map result count to specified maximum" do
      result = Kreuzcrawl.scrape!()
      assert length(result.urls) <= 5
    end
  end

  describe "map_search_filter" do
    test "Filters map results by search keyword" do
      result = Kreuzcrawl.scrape!()
      assert length(result.urls) >= 2
      assert String.contains?(result.urls, "blog")
    end
  end
end

File: e2e/go/map_test.go
// E2e tests for category: map
package e2e_test

import (
	"strings"
	"testing"

	pkg "github.com/kreuzberg-dev/kreuzcrawl"
)

func Test_MapDiscoverUrls(t *testing.T) {
	// Discovers all URLs on a site without fetching full content
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
}

func Test_MapExcludePatterns(t *testing.T) {
	// Excludes URLs matching patterns from URL map
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
}

func Test_MapIncludeSubdomains(t *testing.T) {
	// Includes subdomain URLs in URL map discovery
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
	// skipped: field 'urls' not available on result type
}

func Test_MapLargeSitemap(t *testing.T) {
	// Handles large sitemap with 100+ URLs
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
}

func Test_MapLimitPagination(t *testing.T) {
	// Limits map result count to specified maximum
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
}

func Test_MapSearchFilter(t *testing.T) {
	// Filters map results by search keyword
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
	// skipped: field 'urls' not available on result type
}
